precompiles/utils/src/data.rs

```diff
@@ -607,23 +606,25 @@ impl EvmData for Bytes {
 
 impl EvmData for AccountId32 {
     fn read(reader: &mut EvmDataReader) -> EvmResult<Self> {
-        let range = reader.move_cursor(32)?;
-
-        let data = reader
-            .input
-            .get(range)
-            .ok_or_else(|| revert("tried to parse AccountId32 out of bounds"))?;
+        // AccountId32 is encoded as bytes
```

Comment:
In general I'm not sure that it's correct way, parser doing isn't that we expect of it from the name (definitely static type) and explanation is hidden inside of function body. Let's don't seed future bugs and use `Bytes` parser when we parse argument, AccountId32 is good type for parsing bytes32 IMO

Reply:
I somehowagree with you, although it's grey area with me.

I'd prefer to have parsing as part of `EvmData` instead of custom functions in `dapps-staking-precompiles`.
However, I can change it.

Reply:
@akru reverted to old approach, I was wrong.

It seems that both H160 and AccountId32 are passed as `Bytes`.